sentry/tests/test_tracing.rs

```diff
@@ -106,7 +106,7 @@ fn test_tracing() {
 
 #[tracing::instrument(fields(span_field))]
 fn function() {
-    tracing::Span::current().record("span_field", "some data");
+    tracing::Span::current().record("span_field", &"some data");
```

Comment:
If I remove this, I receive a compilation error:

```
the size for values of type `str` cannot be known at compilation time
the trait `std::marker::Sized` is not implemented for `str`
```

However, the linter now wants to remove this borrow. It looks like we're linting with whatever rust version is installed on the GHA workers without installing a specific version. Is that intended?

Reply:
In CI, this passes again, but locally this continues to give me a compilation error.